### ISSUE TITLE
Screen update rate?  SLEEP_TIME_MSEC default value of 60000 is too large

### ISSUE THREAD

**Issue by vacuumcollapse** [texadactyl/rpi_clock#14]
Hi - rpi3 bullseye with adafruit screen syncing a local ntp server.  Pi is syncing to ntp server accurately, confirmed by `watch -n 0.1 date` and `timedatectl timesync-status` but rpi_clock screen display updates time far less frequency - varies between 2s and 48s every minute, not seemingly dependent on ntp minpoll settings.  Is there a way to set/reduce/manipulate the screen display refresh so it more accurately tracks the ntp system time?  thanks

**Comment by texadactyl** [comment]
@vacumcollapse 

Thanks.  Very interesting. Have not tried this on an adafruit product yet. Never had this symptom on the garbage TFT screen I bought at Amazon. Also, this is not an issue when running with a desktop window display (```FLAG_WINDOWED = True```).

I need to have another look at the source code.  If you see anything, please let me know.  I'll be free soon for investigation.

**Comment by texadactyl** [comment]
@vacumcollapse

Please supply the configuration file that you are using with OWM_API_KEY value masked.  Thanks.

I added a seconds field by changing ```FORMAT_TIME``` to %%I:%%M:%%S

Also, to see if this is hardware related, please try the ```FLAG_WINDOWED = True``` case with a connected HDMI display.  Same behaviour?

**Comment by vacuumcollapse** [comment]
Hi - thanks for looking into this - cool program btw.  Current .cfg attached with details #'d out etc.

Tried it with windowed=true (and big screen via hdmi) and still same behaviour - see screenshot from vnc feed - the time update is ~48s late compared to the system clock.

Does the FORMAT_TIME update need me to reclone or update, or can i just change that in the .cfg?

[rpi_clock.txt](https://github.com/texadactyl/rpi
![Screenshot_20221231_165947](https://user-images.githubusercontent.com/12567316/210150637-9bc88518-8234-4fd0-8c1b-f7593b760c4c.png)
_clock/files/10327669/rpi_clock.txt)

**Comment by vacuumcollapse** [comment]
[rpi_clock.txt](https://github.com/texadactyl/rpi_clock/files/10327693/rpi_clock.txt)

**Comment by vacuumcollapse** [comment]
Confirm with the addtion of :%%S in FORMAT_TIME it shows seconds, but they don't update....

**Comment by vacuumcollapse** [comment]
Some progress:  I noticed the update was happening exactly 60s apart.  In the .cfg, changing SLEEP_TIME_MSEC interval from 60000msec down to 1000msec results in the time updating every 1s onscreen, and the time is aligned to the NTP server derived system time.  However, does this mean it is updating weather every 1s also, which may be a bit much?  thx

**Comment by texadactyl** [comment]
"happening exactly 60s apart"
That is the expected behaviour with the default config file.
Your change to 1000 msecs gave the expected change in behaviour i.e. 1-second tick.
I updated 60000 to 3000 and saw a 3-second tick.
What value to use?  (-:  

It's working on both HDMI and your adafruit device as expected now?
BTW, which adafruit product are you using?  URL, please, if possible.

So, what do you think I should change the SLEEP_TIME_MSEC default value to?

**Comment by vacuumcollapse** [comment]
Well, for an NTP display (use case I was trying to achieve) i'd like it down to 100msec as i have a GPS PPS based NTP server that i know is accurate to a few usec jitter.  But does that timer also control frequency of the OWM web lookup because that doesn't need to be any more regular than e.g. 5mins...  I'm not experienced so couldn't immediately untangle which countdown timer was controlling that lookup

**Comment by texadactyl** [comment]
I'm going to go with a default value of 3000 i.e. a 3-second tick.  I'll highlight this variable in a new CONFIGURATION.md file which will explain all of the parameters.  The operator can experiment as they wish.

**Comment by vacuumcollapse** [comment]
Thanks.  Happy NYE

**Comment by texadactyl** [comment]
Yeah, Happy Occidental New Year!

**Comment by texadactyl** [comment]
@vacumcollapse 
- Which adafruit product are you using?  URL?  3.5" 480x320?
- How did you deviate from the rpi_clock preparation notes?

This one?  https://learn.adafruit.com/adafruit-pitft-3-dot-5-touch-screen-for-raspberry-pi

**Comment by vacuumcollapse** [comment]
Screen was this one https://www.adafruit.com/product/2441

1. Straightforward fresh install of RPIos from the online installer (correcting for the fkms problem with bullseye and older hdmi screens)
2. I followed the adafruit [FBCP install notes](https://learn.adafruit.com/adafruit-pitft-3-dot-5-touch-screen-for-raspberry-pi/easy-install-2) and code for the right screen:
`sudo python3 adafruit-pitft.py --display=35r --rotation=90 --install-type=fbcp` but changed the 90 to 270
3. On your prep notes, I skipped all of it up to "Install RPI Clock" section and it booted up just fine

I believe that was it - it worked first time, and this is all I think i did.
